tests/foreman/cli/test_syncplan.py

```diff
@@ -273,3 +314,99 @@ def test_verify_bugzilla_1261122(self):
+        self.validate_repo_content(
+            repo, ['errata', 'packages'], after_sync=False)
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        self.validate_repo_content(
+            repo, ['errata', 'package-groups', 'packages'])
+
+    @tier2
+    def test_positive_synchronize_products(self):
+        """Create a sync plan with multiple products and sync them
+        automatically.
+
+        @Assert: Products are synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        products = [
+            make_product({'organization-id': self.org['id']})
+            for _ in range(randint(3, 5))
+        ]
+        repos = [
+            make_repository({'product-id': product['id']})
+            for product in products
+            for _ in range(randint(2, 3))
+        ]
+        for repo in repos:
+            self.validate_repo_content(
+                repo, ['errata', 'packages'], after_sync=False)
+        for product in products:
+            Product.set_sync_plan({
+                'id': product['id'],
+                'sync-plan-id': sync_plan['id'],
+            })
+        for repo in repos:
+            self.validate_repo_content(
+                repo, ['errata', 'package-groups', 'packages'])
+
+    @tier4
+    def test_positive_synchronize_future_sync_date(self):
+        """Create a sync plan with sync date in future and sync one product
+        with it automatically.
+
+        @Assert: Product is synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        delay = 10 * 60  # delay for sync date in seconds
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': self.org['id'],
+            'sync-date': (datetime.utcnow() + timedelta(seconds=delay))
+                        .strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': self.org['id']})
+        repo = make_repository({'product-id': product['id']})
+        # Verify product is not synced and doesn't have any content
+        self.validate_repo_content(
+            repo, ['errata', 'packages'], after_sync=False)
+        # Associate sync plan with product
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        # Wait half of expected time
+        sleep(delay/2)
+        # Verify product has not been synced yet
+        self.validate_repo_content(
+            repo, ['errata', 'packages'], after_sync=False)
+        # Wait the rest of expected time
+        sleep(delay/2)
+        # Verify product was synced successfully
+        self.validate_repo_content(
```

Comment:
No need to. `validate_repo_content()` checks for sync status every 30 seconds for 5 minutes by default. The first attempt most likely will fail, but usually second or third one succeeds. Actually that's the reason we introduced `validate_repo_content()` - not to write some workarounds with waiters and loops inside each test.